freshmaker/image_verifier.py

```diff
@@ -21,124 +21,117 @@
         Returns the ContainerRepository object based on the Repository name.
         """
-        query = {
-            "objectType": "containerRepository",
-            "query": {
-                "$and": [
-                    {
-                        "field": "repository",
-                        "op": "=",
-                        "rvalue": repo_name
-                    },
-
-                ]
-            },
-            "projection": [
-                {"field": "*", "include": True, "recursive": True}
-            ]
-        }
+        repos = self.pyxis.find_repositories_by_repository_name(repo_name)
 
-        repos = self.lb.find_container_repositories(query, auto_rebuild=False)
         if not repos:
-            raise ValueError("Cannot get repository %s from Lightblue." % repo_name)
+            raise ValueError("Cannot get repository %s from Pyxis." % repo_name)
+
         if len(repos) != 1:
-            raise ValueError("Multiple records found in Lightblue for repository %s." % repo_name)
+            raise ValueError(
+                "Multiple records found in Pyxis for image repository %s." % repo_name
+            )
 
         return repos[0]
 
-    def _get_repository_from_image(self, nvr):
+    def _get_repository_from_image(self, image):
         """
-        Returns the ContainerRepository object based on the image NVR.
+        Returns the ContainerRepository object based on the image defined by the image NVR.
         """
-        query = {
-            "objectType": "containerRepository",
-            "query": {
-                "$and": [
-                    {
-                        "field": "images.*.brew.build",
-                        "op": "=",
-                        "rvalue": nvr
-                    },
-
-                ]
-            },
-            "projection": [
-                {"field": "*", "include": True, "recursive": True}
-            ]
-        }
+        if "repositories" not in image or not image["repositories"]:
+            raise ValueError("Cannot get repository for image %s from Pyxis." % image["brew"]["build"])
 
-        repos = self.lb.find_container_repositories(query)
-        if not repos:
-            raise ValueError("Cannot get repository for image %s from Lightblue." % nvr)
-        if len(repos) != 1:
-            raise ValueError(
-                "Image %s found in multiple repositories in Lightblue." % nvr)
+        repos = list(filter(lambda x: 'osbs' not in x['repository'], image["repositories"]))
+        image_repo = repos[0]
 
+        if "published" not in image_repo or not image_repo["published"]:
+            for exc in conf.unpublished_exceptions:
+                if image_repo["repository"] == exc["repository"]:
+                    break
+            else:
+                raise ValueError(
+                    "Only images with published repositories or unpublished exceptions can be rebuilt, but this image's repository is not published."
+                )
```

Comment:
We should check against repo data returned from Pyxis's repository API, not the repo data in image, so you should find the repository from Pyxis first then verify it, the data show in image means how this image is present in the repository, not the repository itself.

But since we'll verify the repository data later:
```
def verify_image(...):
    ...
    repo = self._get_repository_from_image(image)                                                                                                                                     
    self._verify_repository_data(repo)
```
it's unnecessary to verify it here.